myapp/app/controllers/tasks_controller.rb

```diff
@@ -2,16 +2,9 @@ class TasksController < ApplicationController
   before_action :fetch_task_by_params_id, only: [:show, :edit, :update, :destroy]
 
   def index
-    @tasks = Task.all
     @task_columns_with_sorting_direction = task_columns_with_sorting_direction
 
-    if params[:name].present? || params[:status].present?
-      filterer = TaskFilterer.new(name: params[:name], status: params[:status])
-      return @tasks unless filterer.valid? # early return if the filterer conditions is invalid
-
-      @tasks = filterer.execute(@tasks)
-    end
-
+    @tasks = Task.status(params[:status]).name_contain(params[:name]).page(params[:page])
     @tasks = @tasks.order("#{params[:sort_key]} #{sort_direction}") if params[:sort_key].present?
```

Comment:
9行目いらなそうに思ったのですが、以下だとどうでしょうか？
```suggestion
    @tasks.order("#{params[:sort_key]} #{sort_direction}") if params[:sort_key].present?
```

Reply:
@masahikokawai 
提案いただいたコードだと order は破壊的なメソッドではないので、 `@tasks` を sort した結果が view に返らないので、sort 機能が機能しなくなりますね。
ただ、 where! にしてあげると動くような気がします！

が、すいません今回はこのまま行かせてください 🙏 
(先STEPも進めておりコンフリクトが怖く… 🙏 )

Reply:
ほんとですね・・失礼しました。。